server/core/store_option.go

```diff
@@ -73,21 +73,30 @@ func SetStoreDeployPath(deployPath string) StoreCreateOption {
 	}
 }
 
-// SetStoreState sets the state for the store.
-func SetStoreState(state metapb.StoreState) StoreCreateOption {
+// OfflineStore offline a store
+func OfflineStore(physicallyDestroyed bool) StoreCreateOption {
 	return func(store *StoreInfo) {
 		meta := proto.Clone(store.meta).(*metapb.Store)
-		meta.State = state
+		meta.State = metapb.StoreState_Offline
+		meta.PhysicallyDestroyed = physicallyDestroyed
 		store.meta = meta
 	}
 }
 
-// OfflineStore offline a store
-func OfflineStore(physicallyDestroyed bool) StoreCreateOption {
+// UpStore up a store
+func UpStore() StoreCreateOption {
 	return func(store *StoreInfo) {
 		meta := proto.Clone(store.meta).(*metapb.Store)
-		meta.State = metapb.StoreState_Offline
-		meta.PhysicallyDestroyed = physicallyDestroyed
+		meta.State = metapb.StoreState_Up
+		store.meta = meta
+	}
+}
+
+// TombstoneStore set a store to tombstone.
+func TombstoneStore() StoreCreateOption {
+	return func(store *StoreInfo) {
+		meta := proto.Clone(store.meta).(*metapb.Store)
+		meta.State = metapb.StoreState_Tombstone
```

Comment:
Should `PhysicallyDestroyed` be changed to `false`? And maybe we should use `func setStoreState(state metapb.StoreState, physicallyDestroyed bool) StoreCreateOption` to reduce redundant code.

Reply:
Agreed, the api design lgtm while we can use `setStoreState` to reduce the duplicated code.

Reply:
As PR #3388 discussed, SetStoreState are replaced by following `StoreCreateOption`s:
```
- UpStore() 
- OfflineStore(physicallyDestroyed bool)
- TombstoneStore()
```
which have clearer definition and logic for each `StoreCreateOption`